### ISSUE TITLE
Moya POST request not working as intended...

### ISSUE THREAD

**Issue by Tritium7** [Moya/Moya#1585]
Hello,

I have a hard time to figure out how should I send a simple POST request with Moya. I spent the last 2 days of testing, but I cant make it work.

The funny thing is, I use Moya for GET requests and those are work perfectly. I manually got the repository for XCode, using : `$ git submodule add https://github.com/Moya/Moya.git`

Here is what I did for POST request:

-I made a 'TargetType' where the 'method' returns '.post'
-I use custom 'parameterEncoding'
-My parameterEncoding looks like:

```swift
public struct SimpleEncoding: ParameterEncoding {
	public static var `default`: SimpleEncoding{ return SimpleEncoding() }

	public func encode(_ urlRequest: URLRequestConvertible, with parameters: Parameters?) throws -> URLRequest {
        	var urlRequest = try urlRequest.asURLRequest()

        	guard let parameters = parameters else { return urlRequest }

        	
            	if urlRequest.value(forHTTPHeaderField: "Content-Type") == nil {
                	urlRequest.setValue("application/x-www-form-urlencoded; charset=utf-8", forHTTPHeaderField: "Content-Type")
            	}

            urlRequest.httpBody = "id=123456" //for testing
        }

        return urlRequest
    }
```

The strange thing is, the Moya logger says I sent a POST request and I saw the desired "id=123456" information in the http body, but the server answers that the 'id' information is missing (the message is the same as when I open the page simply in a browser, because no parameter sent). Also the server administrator said, he saw my tries and looks like I send a GET request instead of POST. (So the response from server is correct, because the information is missing)

The same application works on Android platform using this code:

```swift
String data = "id=123456"
HttpsURLConnection connection = null;

connection = (HttpsURLConnection) url.openConnection();
connection.setReadTimeout(3000);
connection.setConnectTimeout(3000);

connection.setRequestMethod("POST");
connection.setDoInput(true);

//Add our custom CA
connection.setSSLSocketFactory(SslHelper.getMyCustomSslSocket(context));

//Add custom hostname verify
connection.setHostnameVerifier(hostnameVerifier);

//Set the data to send
out = new BufferedOutputStream(connection.getOutputStream());

BufferedWriter writer = new BufferedWriter (new OutputStreamWriter(out, "UTF-8"));

writer.write(data);
writer.flush();

writer.close();
out.close();

 connection.connect();
```

Thank you for your time, and your help in advance!

**Comment by SD10** [comment]
Maybe related to #1533

**Comment by sunshinejr** [comment]
Hey @Tritium7. Can you try to make the same request with `Alamofire`/`URLSession` and see if it works? If yes, please copy-paste the code here so I can prepare a demo project and see what's the problem.

**Comment by Tritium7** [comment]
Hello guys! Sorry for the late answer I was off from work for a week or two.

So finally we figured out what was the issue:

It was a silly mistake, the URL address contained a 'http:' protocoll instead of 'https:' . The funny part is, that everything worked as intended, except the POST requests. 

Finally we discovered the missing 's' letter after we used Wireshark and analyzed the packets. It was an overkill compared the typo problem, but yes we continuously overlooked the address and since the error message from the server wasn't clear I ended up here.

Thanks the quick response to the issue, the problem wasn't on your side!
